src/codeenginestudio/thuy/Numberal.java

```diff
@@ -0,0 +1,21 @@
+package codeenginestudio.thuy;
+
+public abstract class Numberal {
+    int number;
+
+    public abstract Object minus(int i);
+
+    public abstract Object plus(int i);
+
+    public static int parse(String hex, int numberal) {
```

Comment:
Please implement `parse` method in each child class, not implement in here

Reply:
Why not? I edited. I think its OK.